api4/group_test.go

```diff
@@ -8,12 +8,67 @@ import (
 	"net/http"
 	"testing"
 
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+)
 
-	"github.com/stretchr/testify/assert"
+func TestGetGroupsAdminAndNonAdmin(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+	th.App.SetLicense(model.NewTestLicense("ldap"))
+	id := model.NewId()
+	g, err := th.App.CreateGroup(&model.Group{
+		DisplayName: "dn_" + id,
+		Name:        "name" + id,
+		Source:      model.GroupSourceLdap,
+		Description: "description_" + id,
+		RemoteId:    model.NewId(),
+		CreateAt:    model.GetMillis(),
+		UpdateAt:    model.GetMillis(),
+		DeleteAt:    model.GetMillis(),
+	})
+	require.Nil(t, err)
 
-	"github.com/mattermost/mattermost-server/model"
-)
+	opts := model.GroupSearchOpts{
+		IncludeMemberCount: true,
+	}
+
+	a, _ := th.Client.GetGroups(opts)
+	var nonAdminGroup, adminGroup model.Group
```

Comment:
@hahmadia 
Ahh I totally forgot that we had Client4 API to get groups. In this case don't we have the exact same issue where we're returning _all_ the fields and the client has to figure out if a particular value is a real value or the default value which they're getting because they're non-admins. Hmmm... Does non-admins need to ever use this API? If not I would suggest to throw an unauthorized error, and return no data.

Reply:
apologies for missing this part in the first pass 🤦‍♂